addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -328,6 +357,25 @@ function factory(dependencies) {
             inverse: 'attachments',
         }),
         type: attr(),
+        /**
+         * AbortController instance used to upload that attachment to the server.
+         * It's used to have an access to the AbortSignal and allow a control on
+         * the attachment uploading process.
+         */
+        uploadController: attr({
+            compute: '_onIsTemporaryChange',
+            dependencies: ['isTemporary', 'uploadController'],
+        }),
+        /**
+         * AbortSignal instance used to upload that attachment to the server.
+         * The signal allows a control on the uploading process, it's used by
+         * the model to cancel the upload if the attachment is removed before
+         * that process is ended.
+         */
+        uploadSignal: attr({
+            compute: '_computeUploadSignal',
+            dependencies: ['uploadController']
+        }),
```

Comment:
I don't see any purpose for this field

Reply:
I don't think not leaking abort controller is solved by leaking abort controller signal.

The core problem is that the code for attachment upload should be in models, not in component.
I don't really expect this change in this PR, because out of scope of PR.

I think it's fine to leak abort controller, because when this code will move to models, it will use abort controller, which is appropriate.